File: Dogs-C/Dogs-C/Views/JLImageCollectionViewCell.m
//
//  JLImageCollectionViewCell.m
//  Dogs-C
//
//  Created by Jordan Lamb on 10/9/19.
//  Copyright © 2019 Squanto Inc. All rights reserved.
//

#import "JLImageCollectionViewCell.h"
#import "JLBreedController.h"

@implementation JLImageCollectionViewCell

- (void)prepareForReuse
{
    [super prepareForReuse];
    self.dogImageView.image = [UIImage imageNamed:@"dog"];
}

- (void)setImageURL:(NSURL *)imageURL
{
    [JLBreedController.sharedInstance fetchImageData:imageURL completion:^(NSData *imageData, NSError *error) {
        if (error || !imageData) {
            NSLog(@"Error fetching image data %@", error);
            return;
        }
        dispatch_async(dispatch_get_main_queue(), ^{
            self.dogImageView.image = [UIImage imageWithData:imageData];
        });
    }];
}

@end

File: Dogs-C/Dogs-C/Views/JLImageCollectionViewCell.swift
//
//  JLImageCollectionViewCell.swift
//  Dogs-C
//
//  Created by Jordan Lamb on 10/9/19.
//  Copyright © 2019 Squanto Inc. All rights reserved.
//

import UIKit

class JLImageCollectionViewCell: UICollectionViewCell {
    
    var breed: JLBreed? {
        didSet {
            updateViews()
        }
    }
    
    @IBOutlet weak var dogImageView: UIImageView!
    
    func updateViews() {
        
    }
}
